internal/provider/models/alerts/test/threshold_test.go

```diff
@@ -0,0 +1,548 @@
+					}),
+				),
+			},
+
+			// CREATE Threshold Alert for log, minimal configuration
+			{
+				Config: GetCachedConfig(cacheKey) + `
+					resource "mezmo_threshold_alert" "default_log" {
+						pipeline_id = mezmo_pipeline.test_parent.id
+						component_kind = "source"
+						component_id = mezmo_http_source.my_source.id
+						inputs = [mezmo_http_source.my_source.id]
+						name = "my threshold alert"
+						event_type = "log"
+						operation = "custom"
+						script = "function myFunc(a, e, m) { return a }"
+						event_timestamp = ".timestamp"
+						conditional = {
+							expressions = [
+								{
+									field = ".event_count"
+									operator = "greater"
+									value_number = 5000
+								}
+							],
+						}
+						subject = "Threshold Alert for Log event"
+						message = "You received a threshold alert for a Log event"
+						ingestion_key = "abc123"
+					}`,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestMatchResourceAttr(
+						"mezmo_threshold_alert.default_log", "id", regexp.MustCompile(`[\w-]{36}`),
+					),
+					StateHasExpectedValues("mezmo_threshold_alert.default_log", map[string]any{
+						"pipeline_id":                            "#mezmo_pipeline.test_parent.id",
+						"inputs.#":                               "1",
+						"inputs.0":                               "#mezmo_http_source.my_source.id",
+						"active":                                 "true",
+						"component_id":                           "#mezmo_http_source.my_source.id",
+						"component_kind":                         "source",
+						"conditional.%":                          "3",
+						"conditional.expressions.#":              "1",
+						"conditional.expressions.0.field":        ".event_count",
+						"conditional.expressions.0.operator":     "greater",
+						"conditional.expressions.0.value_number": "5000",
+						"conditional.logical_operation":          "AND",
+						"event_type":                             "log",
+						"event_timestamp":                        ".timestamp",
+						"ingestion_key":                          "abc123",
+						"name":                                   "my threshold alert",
+						"operation":                              "custom",
+						"script":                                 "function myFunc(a, e, m) { return a }",
+						"severity":                               "INFO",
+						"style":                                  "static",
+						"subject":                                "Threshold Alert for Log event",
+						"message":                                "You received a threshold alert for a Log event",
+						"window_duration_minutes":                "5",
+						"window_type":                            "tumbling",
+					}),
+				),
+			},
+		},
+	})
+}
+
+// WARNING: Mixing test that throw normal schema validation errors cannot be mixed with
+// tests where we bubble up errors with diag. This tends to cause teardown errors and
+// may be a bug.
```

Comment:
Not sure if this is a bug in our provider or not.  If we have test cases that throw regular validation errors running *after* custom validation tests, it errors in teardown.  The difference is that with schema errors, there won't be a valid `plan`, but our custom errors have one.  Something might be getting muddled with state, but I couldn't track it down.

Reply:
Wow, that's what the problem was? Great investigation!

I guess one could argue that testing the schema definition doesn't need tons of dedicated tests since we trust the TF framework to test that input is validated correctly from the schema. For now it's probably fine to break them into different suites.